modules/core/kivent_core/systems/animation.pyx

```diff
@@ -0,0 +1,118 @@
+            'model': <model-name>,
+            'duration': <duration>
+        }'''
+
+    system_id = StringProperty('animation')
+    processor = BooleanProperty(True)
+    type_size = NumericProperty(sizeof(AnimationStruct))
+    component_type = ObjectProperty(AnimationComponent)
+    system_names = ListProperty(['animation','renderer'])
+
+    def init_component(self, unsigned int component_index,
+                       unsigned int entity_id, str zone, args):
+        model_manager = self.gameworld.model_manager
+        texture_manager = self.gameworld.texture_manager
+
+        cdef MemoryZone memory_zone = self.imz_components.memory_zone
+        cdef AnimationStruct* component = <AnimationStruct*>(
+            memory_zone.get_pointer(component_index))
+        cdef FrameList* frame_list = <FrameList*> animation_manager.animations[args.name]
+        component.entity_id = entity_id
+        component.frames = frame_list
+        component.current_frame = 0
+        component.current_duration = 0
+        component.loop = args.loop
+
+        return self.entity_components.add_entity(entity_id, zone)
+
+    def clear_component(self, unsigned int component_index):
+        cdef MemoryZone memory_zone = self.imz_components.memory_zone
+        cdef AnimationStruct* pointer = <AnimationStruct*>(
+            memory_zone.get_pointer(component_index))
+        pointer.entity_id = -1
+        pointer.frames = NULL
+        pointer.current_frame = 0
+        pointer.current_duration = 0
+        pointer.loop = False
+
+
+    def remove_component(self, unsigned int component_index):
+        cdef AnimationComponent component = self.components[component_index]
+        self.entity_components.remove_entity(component.entity_id)
+        super(AnimationSystem, self).remove_component(component_index)
+
+    def update(self, dt):
+        gameworld = self.gameworld
+        cdef void** component_data = <void**>(
+            self.entity_components.memory_block.data)
+        cdef unsigned int component_count = self.entity_components.count
+        cdef unsigned int count = self.entity_components.memory_block.count
+        cdef unsigned int i, real_index
+        cdef AnimationStruct* anim_comp
+        cdef RenderStruct* render_comp
+
+        for i in range(count):
+            real_index = i*component_count
+            if component_data[real_index] == NULL:
+                continue
+            anim_comp = <AnimationStruct*>component_data[real_index]
+            render_comp = <RenderStruct*>component_data[real_index+1]
+            current_frame = <FrameStruct*>anim_comp.frames[anim_comp.current_frame_index].frame_pointer
+
+            anim_comp.current_duration += dt
+            if current_frame.duration < anim_comp.current_duration:
+                anim_comp.current_duration = 0
+                anim_comp.current_frame_index += 1
+                next_frame = <FrameStruct*>anim_comp.frames[anim_comp.current_frame_index]
+
```

Comment:
This is looking real good so far, here when we change model and texkey being used, we also want to perform manually the same check as done here: https://github.com/kivy/kivent/blob/master/modules/core/kivent_core/systems/renderers.pyx#L154

to make sure we rebatch if it is necessary